dspace-server-webapp/src/test/java/org/dspace/app/rest/DiscoveryScopeBasedRestControllerIT.java

```diff
@@ -0,0 +1,625 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.rest;
+
+import static org.hamcrest.Matchers.containsInAnyOrder;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.is;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import org.dspace.app.rest.matcher.FacetEntryMatcher;
+import org.dspace.app.rest.matcher.FacetValueMatcher;
+import org.dspace.app.rest.test.AbstractControllerIntegrationTest;
+import org.dspace.builder.CollectionBuilder;
+import org.dspace.builder.CommunityBuilder;
+import org.dspace.builder.ItemBuilder;
+import org.dspace.builder.MetadataFieldBuilder;
+import org.dspace.content.Collection;
+import org.dspace.content.Community;
+import org.dspace.content.Item;
+import org.dspace.content.service.CollectionService;
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+
+/**
+ * This class tests the correct inheritance of Discovery configurations for sub communities and collections.
+ * To thoroughly test this, a community and collection structure is set up to where different communities have custom
+ * configurations configured for them.
+ * 
+ * The following structure is uses:
+ * - Parent Community 1     - Custom configuration: discovery-parent-community-1
+ *  -- Subcommunity 11      - Custom configuration: discovery-sub-community-1-1
+ *      -- Collection 111   - Custom configuration: discovery-collection-1-1-1
+ *      -- Collection 112
+ *  -- Subcommunity 12
+ *      -- Collection 121   - Custom configuration: discovery-collection-1-2-1
+ *      -- Collection 122
+ * - Parent Community 2
+ *  -- Subcommunity 21      - Custom configuration: discovery-sub-community-2-1
+ *      -- Collection 211   - Custom configuration: discovery-collection-2-1-1
+ *      -- Collection 212
+ *  -- Subcommunity 22
+ *      -- Collection 221   - Custom configuration: discovery-collection-2-2-1
+ *      -- Collection 222
+ *
+ * Each custom configuration contains a unique index for a unique metadata field, to verify if correct information is
+ * indexed and provided for the different search scopes.
+ *
+ * Each collection has an item in it. Next to these items, there are two mapped items, one in collection 111 and 222,
+ * and one in collection 122 and 211.
+ *
+ * The tests will verify that for each object, the correct facets are provided and that all the necessary fields to
+ * power these facets are indexed properly.
```

Comment:
Can we add a comment to this IT class that it uses/requires the `src/test/data/dspaceFolder/config/spring/api/test-discovery.xml` file?  I think it's important that this IT links back to the test configs, as they are very dependent on one another.